src/datasets/domain/models/NewDataset.ts

```diff
@@ -0,0 +1,21 @@
+import { DatasetLicense } from './Dataset';
+
+export interface NewDataset {
```

Comment:
I agree to rename it as you describe, using the DTO convention.

But I wouldn't place any DTO closer to the use case since it is also used in the repository. In my opinion the best way to locate them is equidistant from the use cases and repositories.

So I created a new folder called "dtos" at the same level as "useCases" and "repositories".